src/docs/deprecation_policy.md

```diff
@@ -1,9 +1,9 @@
 Pants Deprecation Policy
 ========================
 
-For releases after 1.0.0, deprecations are in effect for release branches until the next 2 minor releases (e.g. if the feature is available in 1.0.x it should continue to be available in 1.1.x and 1.2.x and can be removed in 1.3.x).
+For releases after 1.0.0, deprecations are in effect for release branches for the next 5 minor releases (e.g. if a feature is deprecated in 1.0.x it should remain available but deprecated until the 1.5.x stable branch has been cut, and can be removed when master bumps to 1.6.0.dev0).
```

Comment:
I think I understand the general intent here, but part of me wonders if we wouldn't be better off in general leaning more towards a more liberal deprecation policy here (i.e. favoring development velocity over stability) by sticking to 2 minor releases as the minimum. I mainly think this because the primary cases where we bump up against the deprecation policy seem to be on the dev side (cant remove X yet) vs on the consumption side -
 largely because the warnings are pretty noisy and there's always some kind of alternative that can usually be easily adopted in lieu of the deprecated thing. this model might also encourage users to think about deprecations more often (ideally, every release) which seems like it'd be good in general too.

we can still apply our best judgement to the actual deprecations taking place tho to avoid egregious cases, but we'd just have more freedom to keep the codebase clean.

Reply:
Hm. Yea, I'd be fine with that actually.

Interested in what others think: @cheister in particular.

Reply:
In particular: I think you're right that the requirement is less about time, and more about "having one stable release that contains all relevant deprecations", because in the worst case, if someone were 6 months behind, their internal upgrade process would look like bumping through stable releases and fix deprecations until you arrive at the version you want.

Reply:
+1